### PR TITLE
Docker + Update requirements.txt

### PR DESCRIPTION
**Pull request by sirbastiano**
- Updated dependencies --> changed all requirements.txt [Hard dependencies set]
- Update readme.md

NEW FEATURE: Docker Integration with docker-compose and Devcontainers!

- Added dockerfile
- Added docker-compose.yml 
- Added devcontainer.json

### CODE REVIEW

File: phisat-2/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,8 @@
-numpy
-jupyter 
-sentinelhub[AWS]
-eo-learn==1.3.1
+numpy==1.24.4
+h5py==3.9.0
+gdal==3.6.2
```
Comment (devisperessutti) [inline phisat-2/requirements.txt:3]: I would remove `gdal` here, it is not required

File: IMAGIN-e/from-sentinel2/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,8 @@
-eo-learn-core==1.3.1
-eo-learn-io==1.3.1
-h5py
-gdal
+numpy==1.24.4
+h5py==3.9.0
+gdal==3.6.2
```
Comment (devisperessutti) [inline IMAGIN-e/from-sentinel2/requirements.txt:3]: I would remove gdal here, it is not required

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    && bash /tmp/miniconda.sh -b -p $CONDA_DIR \
+    && rm /tmp/miniconda.sh \
+    && $CONDA_DIR/bin/conda update -n base -c defaults conda
+
+
+RUN $CONDA_DIR/bin/conda create -n simulator python=3.8
+# Combine the environment activation and package installations into one RUN command
+RUN echo "source activate simulator" > ~/.bashrc \
+    && /bin/bash -c "source ~/.bashrc \
+    && pip3 install numpy==1.26.4 h5py==3.9.0 jupyter==1.0.0 chardet==4.0.0 opencv-python==4.9.0.80 sentinelhub[AWS]==3.8.3 eo-learn==1.3.1 \
```
Comment (devisperessutti) [inline .devcontainer/Dockerfile:19]: `numpy`'s version should be `1.24.4` (not `1.26.4`)

**Review by devisperessutti — CHANGES_REQUESTED** [review]
Great! Thank you for this very useful addition.

I made minor comments regarding the requirements, and one fix request in the Dockerfile. With the fix I could build and use the Docker image.

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (sirbastiano) [reply .devcontainer/Dockerfile:19]: Agree

**Review by sirbastiano — COMMENTED** [review]
Changes reviewed according to the comments received.

**Comment by sirbastiano** [comment]
Thanks, modified accordingly!Il giorno 18 mar 2024, alle ore 20:01, Devis Peressutti ***@***.***> ha scritto:﻿
@devisperessutti requested changes on this pull request.

Great! Thank you for this very useful addition.
I made minor comments regarding the requirements, and one fix request in the Dockerfile. With the fix I could build and use the Docker image.

In phisat-2/requirements.txt:
> @@ -1,4 +1,8 @@
-numpy
-jupyter 
-sentinelhub[AWS]
-eo-learn==1.3.1
+numpy==1.24.4
+h5py==3.9.0
+gdal==3.6.2

I would remove gdal here, it is not required



In IMAGIN-e/from-sentinel2/requirements.txt:
> @@ -1,4 +1,8 @@
-eo-learn-core==1.3.1
-eo-learn-io==1.3.1
-h5py
-gdal
+numpy==1.24.4
+h5py==3.9.0
+gdal==3.6.2

I would remove gdal here, it is not required



In .devcontainer/Dockerfile:
> +ENV MINICONDA_VERSION 4.9.2
+ENV CONDA_DIR /opt/conda
+ENV PATH $CONDA_DIR/bin:$PATH
+
+# Install anaconda for easy-install of GDAL:
+RUN wget https://repo.anaconda.com/miniconda/Miniconda3-py39_${MINICONDA_VERSION}-Linux-x86_64.sh -O /tmp/miniconda.sh \
+    && bash /tmp/miniconda.sh -b -p $CONDA_DIR \
+    && rm /tmp/miniconda.sh \
+    && $CONDA_DIR/bin/conda update -n base -c defaults conda
+
+
+RUN $CONDA_DIR/bin/conda create -n simulator python=3.8
+# Combine the environment activation and package installations into one RUN command
+RUN echo "source activate simulator" > ~/.bashrc \
+    && /bin/bash -c "source ~/.bashrc \
+    && pip3 install numpy==1.26.4 h5py==3.9.0 jupyter==1.0.0 chardet==4.0.0 opencv-python==4.9.0.80 sentinelhub[AWS]==3.8.3 eo-learn==1.3.1 \

numpy's version should be 1.24.4 (not 1.26.4)

**Comment by sirbastiano** [comment]
Hey Devis, I modified the PR according to your revisions. You can proceed to merge if you agree.

**Comment by devisperessutti** [comment]
Fantastic! Will do. Thanks for the contribution.

**Pull request merged**